hw/drivers/chg_ctrl/adp5061/include/adp5061/adp5061.h

```diff
@@ -49,10 +51,13 @@ struct adp5061_config {
 struct adp5061_dev {
 #if MYNEWT_VAL(BUS_DRIVER_PRESENT)
     struct bus_i2c_node     a_node;
+#if MYNEWT_VAL(ADP5061_USE_CHARGE_CONTROL)
+    struct charge_control   a_chg_ctrl;
+#endif
 #else
     struct os_dev           a_dev;
-#endif
     struct charge_control   a_chg_ctrl;
```

Comment:
So ADP5061_USE_CHARGE_CONTROL setting is only meaningful if BUS_DRIVER_PRESET is set?

Reply:
No, but a_chg_ctrl has cc_itf filed that is used for addressing, type of interface is also defined in charge_control so there is dependency to package and as you noticed, but charge control integration can be disabled, meaning that driver will not register to it.
This is result of having i2c/spi interface settings scattered in different places (sensor/charger maybe more).